lib/ansible/modules/cloud/vmware/vmware_guest_custom_attributes.py

```diff
@@ -219,7 +255,9 @@ def main():
     else:
         # virtual machine does not exists
         module.fail_json(msg="Unable to manage custom attributes for non-existing"
```

Comment:
Funny thing about VMware - does the VM really not exist, or do you just not have permission to see it? :)

Reply:
That might be case!
And it isn't covered by [lib/ansible/module_utils/vmware.py](/ansible/ansible/blob/devel/lib/ansible/module_utils/vmware.py)

Reply:
@jeking3 VMware modules assume you have required permissions and privileges to access environment, because without them modules will not work. We are trying to cover required privileges in [documentation](https://docs.ansible.com/ansible/latest/vmware/scenario_clone_template.html#scenario-requirements)

Reply:
@Akasurde @jeking3 some time ago I've tested a case of not having permissions for custom attributes -> this module emitted `pyVmomi.VmomiSupport.NoPermission: (vim.fault.NoPermission)` error, though I don't remember exactly the place where it was raised.